### ISSUE TITLE
Problem parsing requirement markers

### ISSUE THREAD

**Issue by lvkswrks** [python-poetry/poetry#992]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: OSX
- **Poetry version**: 0.12.11
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/lvkswrks/9967aeedc34f967efeda9c9d429efb5f

## Issue
Poetry seems to have problems parsing requirement markers as illustrated below

	setup(
	    # ...
	    install_requires = [
	        'pyarrow==0.9.0;platform_system!="darwin"',
	        'pyarrow==0.9.post1;platform_system=="darwin"',
	    ],
	)

It will fall into an infinite recursion and crash at some point with

	0: Duplicate dependencies for pyarrow
	0: Different requirements found for pyarrow (0.9.0) and pyarrow (0.9.post1).
	1: Version solving took 0.056 seconds.
	1: Tried 1 solutions.
	0: Retrying dependency resolution for Python (>=3.6,<4.0).
	1: fact: test-multiline is 0.1.0
	1: derived: test-multiline
	1: fact: test-multiline depends on py-setup (*)

Please also see a minimal example project configuration I uploaded to Gist.

**Comment by lvkswrks** [comment]
I tested with poetry 0.12.15 and it is still an issue.

**Comment by lvkswrks** [comment]
The error still persists in Poetry 0.12.17.

```
poetry install -vvv
Using virtualenv: /path/to/.venv
Updating dependencies
Resolving dependencies...
   1: fact: test-multiline is 0.1.0
   1: derived: test-multiline
   1: fact: test-multiline depends on py-setup (*)
   1: selecting test-multiline (0.1.0)
   1: derived: py-setup (*)
   0: Duplicate dependencies for pyarrow
   0: Different requirements found for pyarrow (0.9.0) and pyarrow (0.9.post1).
   1: Version solving took 0.221 seconds.
   1: Tried 1 solutions.
   0: Retrying dependency resolution for Python (>=3.6,<4.0).
   1: fact: test-multiline is 0.1.0
   1: derived: test-multiline
   1: fact: test-multiline depends on py-setup (*)
   1: selecting test-multiline (0.1.0)
   1: derived: py-setup (*)
   0: Duplicate dependencies for pyarrow
   0: Different requirements found for pyarrow (0.9.0) and pyarrow (0.9.post1).
   1: Version solving took 0.004 seconds.
   1: Tried 1 solutions.
```

**Comment by lvkswrks** [comment]
Hi, the issue persists in 1.0.0.b.

**Comment by s0undt3ch** [comment]
The issue still exists in latest release, see https://github.com/python-poetry/poetry/issues/1238#issuecomment-575608797

**Comment by s0undt3ch** [comment]
Though, I hit a Max recursion error.

**Comment by g4b1nagy** [comment]
This has been fixed in the `1.1.X` branch of `Poetry`. So every version starting from `1.1.0` (Oct 1, 2020) should work just fine.
Note that:
> lock files generated with this release are not compatible with previous releases of Poetry.
